x-pack/plugins/apm/public/components/app/ServiceOverview/__test__/ServiceOverview.test.js

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { shallow } from 'enzyme';
+import ServiceOverview from '../view';
+import { STATUS } from '../../../../constants';
+import * as apmRestServices from '../../../../services/rest/apm';
+
+jest.mock('../../../../services/rest/apm');
+
+describe('Service Overview -> View', () => {
+  let wrapper;
+  let instance;
+
+  beforeEach(() => {
+    wrapper = shallow(<ServiceOverview serviceList={{}} />);
+    instance = wrapper.instance();
+  });
+
+  it('should render when historical data is found', () => {
+    expect(wrapper).toMatchSnapshot();
+    const List = wrapper
+      .find('ServiceListRequest')
+      .props()
+      .render({});
+    expect(List.props).toMatchSnapshot();
+  });
+
+  it('should render when historical data is not found', () => {
+    wrapper.setState({ historicalDataFound: false });
+    expect(wrapper).toMatchSnapshot();
+    const List = wrapper
+      .find('ServiceListRequest')
+      .props()
+      .render({});
+    expect(List.props).toMatchSnapshot();
+  });
+
+  describe('checking for historical data', () => {
+    let mockAgentStatus;
+
+    beforeEach(() => {
+      mockAgentStatus = {
+        dataFound: true
+      };
+      // eslint-disable-next-line import/namespace
+      apmRestServices.loadAgentStatus = jest.fn(() =>
+        Promise.resolve(mockAgentStatus)
+      );
+    });
+
+    it('should happen if service list status is success and data is empty', async () => {
+      const props = {
+        serviceList: {
+          status: STATUS.SUCCESS,
+          data: []
+        }
+      };
+      await instance.checkForHistoricalData(props);
+      expect(apmRestServices.loadAgentStatus).toHaveBeenCalledTimes(1);
```

Comment:
In other tests I have use `wrapper.setProps(props)`. I did that to also test the lifecycle methods. However, this looks really clean and might be a better approach. 
What do you think?

Reply:
Yeah, I originally did that too. In this case, since the instance method receives the props as an argument, I think it's nice to test it directly. I thought about also spying on the instance method and then just making sure it's called on mount and update, but I couldn't think of a clean way to do that for `componentDidMount` because I can't spy on it until after it's mounted, so I just skipped.

Reply:
👍